src/librustdoc/html/static/main.js

```diff
@@ -812,9 +880,38 @@
                 if (resultIndex[i].id > -1) {
                     obj = searchIndex[resultIndex[i].id];
                     filterdata.push([obj.name, obj.ty, obj.path, obj.desc]);
-                    results.push(obj);
+                    if (!obj.type) {
+                        if (results['others'].length < maxResults) {
+                            results['others'].push(obj);
+                        }
+                    } else {
```

Comment:
Forgive my ignorance of the JS part of rustdoc, but i think i'm missing something. What is `obj.type` here, and why does its presence keep it from getting into `results['others']`? Is that on purpose?

Reply:
You can see everything as a hashmap in js, so `obj.type` == `obj['type']`. In this case, if there is no `type` (which is just a regular field, nothing particular), it means it's not a function nor a method so I put it in "others".

Reply:
So is `results['others']` not meant to be the same as the current search results? I'm concerned that you won't be able to find a function by name if it's not returning something with the same name. For example, are you able to find `str::to_lowercase` by searching for something like `lowercase`? Will you be able to find that method on any tab in your new layout?